libexec/python/shell.py

```diff
@@ -167,15 +258,8 @@ def parse_image_uri(image,
     parsed = {'registry': registry,
               'namespace': namespace,
               'repo_name': repo_name,
-              'repo_tag': repo_tag}
-
-    # No field should be empty
-    for fieldname, value in parsed.items():
-        if len(value) == 0:
-            bot.error("%s found empty, check uri! Exiting." % value)
-            sys.exit(1)
-
-    # Version is not required
-    parsed['version'] = version
+              'repo_tag': repo_tag,
+              'version': version,
+              }
 
```

Comment:
stylistic thing, can we keep:

```
    parsed = { 'registry': registry,
                      'namespace': namespace,
                      'repo_name': repo_name,
                      'repo_tag': repo_tag,
                      'version': version }
```

Reply:
You mean the closing brace `}` on the same line as the last item? Sure, if that's the preferred style. Normally I write it that way to make it easier to add items to the end of the list (simply copy and paste the last line), but that's not a problem, I'll change it.

Reply:
Thanks!